### PR TITLE
[LoongArch64] delete the rsGetRsvdReg() for LA64.

### PR DESCRIPTION
**Pull request by shushanhf**
This PR is part of the issue https://github.com/dotnet/runtime/issues/69705 to amend the LA's port.

Delete the rsGetRsvdReg() for LA64 because LA64 had reserved the REG_R21.

This is only for optimization.
Had tested this PR on LA64.

### CODE REVIEW

**Comment by shushanhf** [comment]
Hi, @jakobbotsch 
Could you please review this PR?
Thanks

**Comment by shushanhf** [comment]
> Hi, @jakobbotsch Could you please review this PR? Thanks

Hi, @jakobbotsch Could you please review this PR?
Next week I will push some other PRs.
Thanks

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -4068,8 +4068,8 @@ bool Compiler::compRsvdRegCheck(FrameLayoutState curState)
     return true; // just always assume we'll need it, for now
 
 #elif defined(TARGET_LOONGARCH64)
-    JITDUMP(" Returning true (LOONGARCH64)\n\n");
-    return true; // just always assume we'll need it, for now
+    JITDUMP(" Returning false (LOONGARCH64)\n\n");
+    return false; // LoongArch64 reserved other special REG_R21.
```
Comment (jakobbotsch) [inline src/coreclr/jit/compiler.cpp:4072]: This is dead code now, just delete  it.

Comment (shushanhf) [reply src/coreclr/jit/compiler.cpp]: Ok, Thanks

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**